### ISSUE TITLE
Expose a callback for onFocus in SemanticsProperties

### ISSUE THREAD

**Issue by lorrainekan** [flutter/flutter#14424]
@goderbauer 

I want to know when a particular node built by overriding semanticsBuilder in CustomPaint has focus (and not require the user to have to focus and then tap - so onTap doesn't work for my use case). Please add a callback for onFocus.

**Comment by goderbauer** [comment]
Just to avoid any misunderstandings: We are talking about accessibility focus here, right? The green (Android) or black (iOS) square that Talkback/VoiceOver shows on screen to allow a user to interact with the node?

This is not about input focus (e.g. a text field has input focus when it receives keyboard inputs).

**Comment by lorrainekan** [comment]
Yes, this is about the accessibility focus - the green (Android) square that shows on the screen when Talkback is enabled and the user is focused on a node. When testing on Android, Talkback will focus on a node when I tap one or when I swipe left or right to iterate the nodes. In my CustomPainter, I am generating a list of nodes and I want to know when that node has the Talkback focus.
